### DISCUSSION TITLE
OAUTH1 Signature Method: HMAC-SHA256

### DISCUSSION THREAD

**Discussion by Magayver89** [viniciussanchez/RESTRequest4Delphi#199] [Q&A]
O RestRequest4Delphi tem suporte para este tipo de conexão?
OAUTH1
Signature Method: HMAC-SHA256
utilizando: Consumer Key, Consumer Secret, Access Token,Token Secret

**Comment by viniciussanchez** [comment]
Como você faria isso no Postman? tem documentação?

**Reply by Magayver89** [reply]
![image](https://github.com/viniciussanchez/RESTRequest4Delphi/assets/143232642/2b2975ca-27c2-4db3-9e40-fb0e2ad85a3d)

**Reply by Magayver89** [reply]
curl --location 'https://magayvertech.com.br/index.php/rest/V1/categories' \
--header 'Authorization: OAuth oauth_consumer_key="xhi53ezz9nf6nd07fmw75g7vb1dcn6vm",oauth_token="80lgp6kkbf200she5wwfynpkcjbkcsj3",oauth_signature_method="HMAC-SHA256",oauth_timestamp="1712239049",oauth_nonce="uvzCTPOyxMI",oauth_version="1.0",oauth_signature="7Dsm%2F8SwXqqOwyXBpQHBSdTEOLDoKPJMtyM5ub4sRmc%3D"' \
--header 'Cookie: PHPSESSID=9eda6a95dc2923f7ef7488a92966e714'

**Reply by viniciussanchez** [reply]
Imagino que dê para você adicionar o header no request passando esse conteúdo...

**Reply by Magayver89** [reply]
Poderia me passar um exemplo de como ficaria?

**Reply by Magayver89** [reply]
![image](https://github.com/viniciussanchez/RESTRequest4Delphi/assets/143232642/f2f67518-59b6-47cf-b4af-9c9cc8200151)

**Reply by Magayver89** [reply]
![image](https://github.com/viniciussanchez/RESTRequest4Delphi/assets/143232642/bfbb9abd-6eea-4278-940d-2185c4b2be47)

**Reply by viniciussanchez** [reply]
Ao meu ver, teria que ser apenas um AddHeader, assim:

AddHeader('Authorization', 'OAuth oauth_consumer_key="xhi53ezz9nf6nd07fmw75g7vb1dcn6vm",oauth_token="80lgp6kkbf200she5wwfynpkcjbkcsj3",oauth_signature_method="HMAC-SHA256",oauth_timestamp="1712239049",oauth_nonce="uvzCTPOyxMI",oauth_version="1.0",oauth_signature="7Dsm%2F8SwXqqOwyXBpQHBSdTEOLDoKPJMtyM5ub4sRmc%3D"')

E mais um para o cookie

AddHeader('Cookie', 'PHPSESSID=9eda6a95dc2923f7ef7488a92966e714')

**Reply by viniciussanchez** [reply]
Talvez teria que acrescentar no final do AddHeader o poDoNotEncode
